File: samples/spring-mvc/src/main/java/io/openapiprocessor/samples/FooController.java
package io.openapiprocessor.samples;

import io.openapiprocessor.openapi.api.FooApi;
import io.openapiprocessor.openapi.model.Foo;
import org.springframework.web.bind.annotation.RestController;

import java.util.UUID;

@RestController
public class FooController implements FooApi {

    @Override
    public Foo postFoo(Foo foo) {
        return foo != null ? foo : new Foo("foo", UUID.randomUUID(), new MappedBar());
    }
}

File: samples/spring-mvc-boot-4-packages-kt/src/main/kotlin/io/openapiprocessor/samples/foo/FooController.kt
/*
 * Copyright 2015 https://github.com/openapi-processor/openapi-processor-samples
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiprocessor.samples.foo

import org.springframework.web.bind.annotation.RestController
import java.util.UUID

@RestController
class FooController: FooApi {
    override fun postFoo(foo: Foo?): Foo {
        return foo ?: Foo("foo", UUID.randomUUID())
    }
}
